test/language/expressions/exponentiation/exp-operator-precedence-unary-expression-semantics.js

```diff
@@ -0,0 +1,63 @@
+// Copyright (C) 2015 Rick Waldron. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+author: Rick Waldron
+email: waldron.rick@gmail.com
+es7id: pending
+description: >
+  ExponentiationExpression :
+    UnaryExpression
+    ...
+
+  UnaryExpression :
+    ...
+    `delete` UnaryExpression
+    `void` UnaryExpression
+    `typeof` UnaryExpression
+    `+` UnaryExpression
+    `-` UnaryExpression
+    `~` UnaryExpression
+    `!` UnaryExpression
+---*/
+
+assert.sameValue(-(3 ** 2), -9);
+assert.sameValue(+(3 ** 2), 9);
+assert.sameValue(~(3 ** 2), -10);
+assert.sameValue(!(3 ** 2), false);
+
+
+assert.sameValue(2 ** -2, 0.25);
+
+var o = { p: 1 };
+// delete o.p -> true -> ToNumber(true) -> 1
+assert.sameValue(2 ** delete o.p, 2);
+
+// void 1 -> undefined -> ToNumber(undefined) -> NaN
+assert.sameValue(2 ** void 1, NaN);
+
+// typeof 1 -> "number" -> ToNumber("number") -> NaN
+assert.sameValue(2 ** typeof 1, NaN);
+
+var s = "2";
+var n = 2;
+
+// +s -> +"2" -> 2 -> ToNumber(2) -> 2
+assert.sameValue(2 ** +s, 4);
```

Comment:
This makes use of the ToNumber() from https://tc39.github.io/ecma262/#sec-unary-plus-operator-runtime-semantics-evaluation rather than the ToNumber() in http://rwaldron.github.io/exponentiation-operator/#exp-operator-runtime-semantics-evaluation --- is that intentional?